evap/staff/importers.py

```diff
@@ -97,28 +98,98 @@ def store_in_database(self, vote_start_datetime, vote_end_date, semester):
         assert not self.errors
         # This is safe because the user's email address is checked before in the importer (see #953)
         responsible_dbobj = UserProfile.objects.get(email=self.responsible_email)
-        course = Course(
-            name_de=self.name_de,
-            name_en=self.name_en,
-            type=self.course_type,
+        if self.existing_course:
+            self.existing_course.degrees.add(*self.degrees)
+        else:
+            course = Course(
+                name_de=self.name_de,
+                name_en=self.name_en,
+                type=self.course_type,
+                semester=semester,
+            )
+            course.save()
+            course.responsibles.set([responsible_dbobj])
+            course.degrees.set(self.degrees)
+            evaluation = Evaluation(
+                vote_start_datetime=vote_start_datetime,
+                vote_end_date=vote_end_date,
+                course=course,
+                wait_for_grade_upload_before_publishing=self.is_graded,
+            )
+            evaluation.save()
+            evaluation.contributions.create(
+                evaluation=evaluation,
+                contributor=responsible_dbobj,
+                role=Contribution.Role.EDITOR,
+                textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
+            )
+
+    def check_existing_course(self, semester, importer):
+        name_collision = self.check_name_collision(semester)
+        if name_collision:
+            # Only if there are name collisions, it is worth to check for an existing course
+            if self.find_and_set_existing_course(semester, importer):
+                return True
+            if name_collision == "both":
+                importer.errors[ImporterError.COURSE].append(
+                    _("Course {} ({}) does already exist in this semester.").format(self.name_en, self.name_de)
+                )
+            else:
+                importer.errors[ImporterError.COURSE].append(
+                    _("Course {} does already exist in this semester.").format(
+                        self.name_en if name_collision == "name_en" else self.name_de
+                    )
+                )
+        return False
```

Comment:
I can't really follow what the returned value means from just reading this method, I would personally write it more like this, but this really is more of a personal preference. (this doesn't factor in any other comments I wrote)
```python
name_collision = self.check_name_collision(semester)
if not name_collision:
    return False

self.find_and_set_existing_course(semester, importer)  # (returns nothing)
if self.existing_course is None:
    if name_collision == "both":
        importer.errors[ImporterError.COURSE].append(
            _("Course {} ({}) does already exist in this semester.").format(self.name_en, self.name_de)
        )
    else:
        importer.errors[ImporterError.COURSE].append(
            _("Course {} does already exist in this semester.").format(getattr(self, name_collision))
        )

return self.existing_course is not None
```

Reply:
Yeah its a more clearer structure, I like it